classes/app.rb

```diff
@@ -37,67 +26,71 @@ def list_people
   end
 
   # Create a student
-  def create_student
-    print 'Age: '
-    age = gets.to_i
-    print "\nName: "
-    name = gets.chomp.capitalize
-    print "\nHas parent permission? [Y/N]: "
+  def create_student(name, age)
+    print 'Has parent permission? [Y/N]: '
     permission = gets.downcase == 'y'
     @people << Student.new(age, name, permission)
     puts 'Person created successfully'
   end
 
   # Create a teacher
-  def create_teacher
-    print 'Age: '
-    age = gets.to_i
-    print "\nName: "
-    name = gets.chomp.capitalize
-    print "\nSpecialization: "
+  def create_teacher(name, age)
+    print 'Specialization: '
     specialization = gets.chomp.capitalize
     @people << Teacher.new(specialization, age, name)
     puts 'Person created successfully'
   end
 
   # option 3 - Create a person
   def create_person
-    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
     option = gets.chomp
+    print 'Name: '
+    name = gets.chomp.capitalize
+    print 'Age: '
+    age = gets.to_i
```

Comment:
Nice Job so far👍🏼. In obeying the Single responsibility rule as required, I noticed you have assigned the responsibility of input validation and request to the `main.rb` and as regards the rule, a class should have one responsibility👍🏼. But I have noticed that you are still assigned the responsibility of requesting and validating input in the app class, which defeats the rule😉. Kindly do well to ensure all commands/methods regarding input requests and validation are taken to the `main.rb` file to stay consistent with the single responsibility rule👍🏼.

you can make use of [this lesson](https://www.digitalocean.com/community/conceptual-articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design) for clarity

Reply:
@Ibizugbe  Done making changes